ansible/library/apply_device_conn_sonic.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+from ansible.module_utils.basic import AnsibleModule
+import traceback
+import subprocess
+import shlex
+
+
+def get_port_lists():
+    cmd = "show interfaces description | grep Ethernet | awk '{print $1}'"
+    ps = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = ps.communicate()[0].decode('utf-8')
+    port_list = [line for line in output.split('\n') if line.strip()]
+    return port_list
+
+
+def get_platform():
+    cmd = "redis-cli --raw -n 4 hget 'DEVICE_METADATA|localhost' 'platform'"
+    ps = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = ps.communicate()[0].decode('utf-8')
+    return output.strip()
+
+
+def set_autoneg(port, value):
+    if value == "on":
+        value = "enabled"
+    elif value == "off":
+        value = "disabled"
+    else:
+        raise Exception("Invalid autoneg value: %s" % value)
+    cmd = "config interface autoneg %s %s" % (port, value)
+    ps = subprocess.Popen(shlex.split(cmd), shell=False, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    (_, stderr) = ps.communicate()
+    if ps.returncode != 0:
+        if stderr:
+            stderr = stderr.decode('utf-8')
+        raise Exception("Failed to set autoneg for port %s: %s" % (port, stderr))
+
+
+def apply_device_conn(module, platform, port_list, device_conn):
+    # Apply Autoneg
+    default_autoneg_value = None
+    # For SP4 and SP3, the default autoneg value is "on"
+    if "nvidia_sn5" in platform or "nvidia_sn4" in platform:
+        default_autoneg_value = "on"
```

Comment:
Suggest removing this hardcoded logic and using link definition instead. Otherwise it can be a little confusing when the value in link definition is not used.

Reply:
Yes, we will use the defined value first. Only if this value isn't in the links.csv. We will use this default value.